services/src/error.rs

```diff
@@ -491,6 +491,14 @@ pub enum Error {
         resource_type: String,
         resource_id: String,
     },
+
+    ExpirationTimestampInPast,
+    IllegalExpirationUpdate {
+        reason: String,
+    },
+    IllegalDatasetStatus {
+        status: String,
```

Comment:
add dataset it and snafu display

Reply:
265eba09b4725423b2064fad2ad936bbf66f69c0